service/controller/app/v1/appcatalog/appcatalog.go

```diff
@@ -63,7 +59,7 @@ func (a *AppCatalog) GetCatalogForApp(ctx context.Context, customResource v1alph
 
 	a.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("looking for appCatalog %#q in namespace %#q", catalogName, a.watchNamespace))
 
-	appCatalog, err := a.g8sClient.ApplicationV1alpha1().AppCatalogs(a.watchNamespace).Get(catalogName, metav1.GetOptions{})
+	appCatalog, err := a.g8sClient.ApplicationV1alpha1().AppCatalogs("default").Get(catalogName, metav1.GetOptions{})
```

Comment:
If we do not set it as `default`, it failed as below. 
```
E 04/04 13:34:13 /apis/application.giantswarm.io/v1alpha1/namespaces/giantswarm/apps/test-app stop reconciliation due to error | operatorkit/controller/controller.go:347 | controller=app-operator | event=update | loop=1 | version=4254
	/Users/jihyukbok/go/src/github.com/giantswarm/app-operator/vendor/github.com/giantswarm/operatorkit/controller/resource_set.go:72:
	/Users/jihyukbok/go/src/github.com/giantswarm/app-operator/service/controller/app/v1/resource_set.go:204:
	/Users/jihyukbok/go/src/github.com/giantswarm/app-operator/service/controller/app/v1/appcatalog/appcatalog.go:66:
	an empty namespace may not be set when a resource name is provided
```
Anyway `appcatalog` is a cluster-based scope resource, so it doesn't matter.

Reply:
I think this might be an issue in our library there as we should not require namespaces for non-namespaced resources. Your trick here is a quick workaround but we should fix this in general, maybe sth to talk with @giantswarm/sig-operator about

Reply:
I think thatmight be what @kopiczko linked to below

Reply:
So no bug

Reply:
This is outdated & fixed already.